odoo/models.py

```diff
@@ -64,6 +64,13 @@
 from .tools.misc import CountingStream, LastOrderedSet, ReversedIterable, unquote
 from .tools.translate import _, _lt
 
+import typing
+if typing.TYPE_CHECKING:
+    from collections.abc import Reversible
+    from .modules.registry import Registry
+from odoo.api import Self, ValuesType, IdType
```

Comment:
Used only for type checking.
```suggestion
    from odoo.api import Self, ValuesType, IdType
```